app/assets/javascripts/controllers/floating_ip/floating_ip_form_controller.js

```diff
@@ -1,34 +1,43 @@
-ManageIQ.angular.app.controller('floatingIpFormController', ['$http', '$scope', 'floatingIpFormId', 'miqService', function($http, $scope, floatingIpFormId, miqService) {
+ManageIQ.angular.app.controller('floatingIpFormController', ['floatingIpFormId', 'miqService', 'API', function(floatingIpFormId, miqService, API) {
   var vm = this;
-  vm.floatingIpModel = { floating_ip_address: '' };
-  vm.formId = floatingIpFormId;
-  vm.afterGet = false;
-  vm.modelCopy = angular.copy( vm.floatingIpModel );
-  vm.model = "floatingIpModel";
 
-  ManageIQ.angular.scope = vm;
-  vm.saveable = miqService.saveable;
+  var init = function() {
+    vm.afterGet = false;
 
-  if (floatingIpFormId == 'new') {
-    vm.floatingIpModel.floating_ip_address = "";
-    vm.floatingIpModel.description = "";
-    vm.newRecord = true;
-  } else {
-    miqService.sparkleOn();
+    vm.floatingIpModel = { address: null };
+    vm.available_ems = null;
+
+    vm.formId = floatingIpFormId;
+    vm.model = "floatingIpModel";
+    vm.newRecord = floatingIpFormId === "new";
+    vm.saveable = miqService.saveable;
 
-    $http.get('/floating_ip/floating_ip_form_fields/' + floatingIpFormId)
-      .then(getFloatingIpFormData)
-      .catch(miqService.handleFailure);
-  }
+    miqService.sparkleOn();
+    if (vm.newRecord) {
+      API.get("/api/providers?expand=resources&attributes=name&filter[]=type='*NetworkManager'").then(function(data) {
+        vm.available_ems = data.resources;
+        vm.afterGet = true;
+        vm.modelCopy = angular.copy(vm.floatingIpModel);
+        miqService.sparkleOff();
+      }).catch(miqService.handleFailure);
+    } else {
+      API.get("/api/floating_ips/" +  floatingIpFormId + "?attributes=cloud_network,cloud_tenant,ext_management_system,network_port").then(function(data) {
+        Object.assign(vm.floatingIpModel, data);
+        vm.afterGet = true;
+        vm.modelCopy = angular.copy(vm.floatingIpModel);
+        miqService.sparkleOff();
+      }).catch(miqService.handleFailure);
+    }
+  };
 
   vm.addClicked = function() {
-    var url = 'create/new' + '?button=add';
+    var url = 'create/new?button=add';
```

Comment:
This should be `'/floating_ip/create/new?button=add'`  as below in `vm.cancelClicked`. That way you don't assume it's gonna be `/floating_ip/` + `url`.

Reply:
Sure.